torch/_dynamo/allowed_functions.py

```diff
@@ -253,6 +254,16 @@ def is_allowed(obj):
     # torch.ops is populated lazily so we don't necessarily have them in
     # _allowed_function_ids.  Figure it out by testing the type instead
     # in those cases
+    if obj in (
```

Comment:
@anijain2305 I didn't want to have to modify `is_allowed` function at all, so I didn't worry about doing the modification in a clean/maintainable way.  I want to find a way to inline all these functions.  What's the best way to do this?

I think it's cleaner from readability perspective if I apply a decorator inside _functional_collectives.py on each of these functions, as that will enable anyone adding new functions to discover/copy/paste and this way is easy to miss.

Reply:
This diff works 

~~~
diff --git a/torch/_dynamo/allowed_functions.py b/torch/_dynamo/allowed_functions.py
index d81a2ab0f87..4fe693a64f1 100644
--- a/torch/_dynamo/allowed_functions.py
+++ b/torch/_dynamo/allowed_functions.py
@@ -109,22 +109,9 @@ def _disallowed_function_ids():
         torch._C._dynamo.eval_frame.unsupported,
     ]

-    try:
+    if torch.distributed.is_available():
         from torch.distributed import _functional_collectives
-
-        remove.extend(
-            [
-                _functional_collectives.all_reduce,
-                _functional_collectives.all_gather_tensor,
-                _functional_collectives.reduce_scatter_tensor,
-                _functional_collectives._expand_group,
-                _functional_collectives._maybe_wrap_tensor,
-                _functional_collectives._are_we_tracing,
-            ]
-        )
-        config.skipfiles_inline_module_allowlist.append(_functional_collectives)
-    except ImportError:
-        pass
+        config.skipfiles_inline_module_allowlist.add(_functional_collectives)

     # extract all dtypes from torch
     dtypes = [
@@ -164,7 +151,6 @@ def _allowed_function_ids():
             "torch._C.inductor.",
             "torch.fx.",
             "torch.distributed.fsdp.",
-            "torch.distributed._functional_collectives.",
         )
         allowed_modules_dot = tuple([x + "." for x in allowed_modules])
         module = inspect.getmodule(obj)
~~~

The generated Dynamo graphs look like these

~~~
 ===== __compiled_fn_2 =====
 <eval_with_key>.5 class GraphModule(torch.nn.Module):
    def forward(self, L_inp_ : torch.Tensor):
        l_inp_ = L_inp_

        # File: /scratch/anijain/work/pytorch/torch/distributed/_functional_collectives.py:432, code: tensor = torch.ops.c10d_functional.reduce_scatter_tensor(self, reduceOp, tag, rankset, group_size)  # type: ignore[attr-defined]
        reduce_scatter_tensor = torch.ops.c10d_functional.reduce_scatter_tensor(l_inp_, 'sum', '', [0], 1);  l_inp_ = None

        # File: /scratch/anijain/work/pytorch/torch/distributed/_functional_collectives.py:345, code: return torch.ops.c10d_functional.wait_tensor(tensor)  # type: ignore[attr-defined]
        wait_tensor = torch.ops.c10d_functional.wait_tensor(reduce_scatter_tensor);  reduce_scatter_tensor = None
        return (wait_tensor,)
        
~~~

Reply:
Yep, that works. Thanks @anijain2305!